### ISSUE TITLE
Peeker times out waiting for Kafka topics during benchmark startup

### ISSUE THREAD

**Issue by cirego** [MaterializeInc/materialize#4496]
I've been running into an intermittent issue where Materialize never starts ingesting records during the benchmark run. It appears that it's due to a problem where Peeker cannot create the sources, which may be due to a timing issue with querying the confluent registry.

Here are the logs from a failed Peeker container:
```
[2020-10-14T04:04:32Z INFO  peeker] installed source item for topic debezium.tpcch.item                                                                                                                                                        
[2020-10-14T04:04:32Z WARN  peeker] error trying to create source nation: db error: ERROR: fetching latest schema for subject 'debezium.tpcch.nation-value' from registry: subject not found    
[2020-10-14T04:04:32Z WARN  peeker] error trying to create source stock: db error: ERROR: fetching latest schema for subject 'debezium.tpcch.stock-value' from registry: subject not found  
[2020-10-14T04:04:32Z WARN  peeker] error trying to create source supplier: db error: ERROR: fetching latest schema for subject 'debezium.tpcch.supplier-value' from registry: subject not found
[2020-10-14T04:04:32Z WARN  peeker] error trying to create source nation: db error: ERROR: fetching latest schema for subject 'debezium.tpcch.nation-value' from registry: subject not found    
[2020-10-14T04:04:32Z WARN  peeker] error trying to create source region: db error: ERROR: fetching latest schema for subject 'debezium.tpcch.region-value' from registry: subject not found
[2020-10-14T04:04:32Z WARN  peeker] error trying to create source supplier: db error: ERROR: fetching latest schema for subject 'debezium.tpcch.supplier-value' from registry: subject not found
[2020-10-14T04:04:35Z WARN  peeker] error trying to create source nation: db error: ERROR: fetching latest schema for subject 'debezium.tpcch.nation-value' from registry: subject not found    
[2020-10-14T04:04:35Z WARN  peeker] error trying to create source stock: db error: ERROR: fetching latest schema for subject 'debezium.tpcch.stock-value' from registry: subject not found  
[2020-10-14T04:04:35Z WARN  peeker] error trying to create source supplier: db error: ERROR: fetching latest schema for subject 'debezium.tpcch.supplier-value' from registry: subject not found
[2020-10-14T04:04:35Z WARN  peeker] error trying to create source nation: db error: ERROR: fetching latest schema for subject 'debezium.tpcch.nation-value' from registry: subject not found    
[2020-10-14T04:04:35Z WARN  peeker] error trying to create source region: db error: ERROR: fetching latest schema for subject 'debezium.tpcch.region-value' from registry: subject not found
[2020-10-14T04:04:35Z WARN  peeker] error trying to create source supplier: db error: ERROR: fetching latest schema for subject 'debezium.tpcch.supplier-value' from registry: subject not found
[2020-10-14T04:04:38Z WARN  peeker] error trying to create source nation: db error: ERROR: fetching latest schema for subject 'debezium.tpcch.nation-value' from registry: subject not found    
[2020-10-14T04:04:38Z WARN  peeker] error trying to create source stock: db error: ERROR: fetching latest schema for subject 'debezium.tpcch.stock-value' from registry: subject not found  
[2020-10-14T04:04:38Z WARN  peeker] error trying to create source supplier: db error: ERROR: fetching latest schema for subject 'debezium.tpcch.supplier-value' from registry: subject not found
[2020-10-14T04:04:38Z WARN  peeker] error trying to create source nation: db error: ERROR: fetching latest schema for subject 'debezium.tpcch.nation-value' from registry: subject not found    
[2020-10-14T04:04:38Z WARN  peeker] error trying to create source region: db error: ERROR: fetching latest schema for subject 'debezium.tpcch.region-value' from registry: subject not found    
[2020-10-14T04:04:38Z WARN  peeker] error trying to create source supplier: db error: ERROR: fetching latest schema for subject 'debezium.tpcch.supplier-value' from registry: subject not found
[2020-10-14T04:04:41Z WARN  peeker] error trying to create source nation: db error: ERROR: fetching latest schema for subject 'debezium.tpcch.nation-value' from registry: subject not found    
[2020-10-14T04:04:41Z WARN  peeker] error trying to create source stock: db error: ERROR: fetching latest schema for subject 'debezium.tpcch.stock-value' from registry: subject not found      
[2020-10-14T04:04:41Z WARN  peeker] error trying to create source supplier: db error: ERROR: fetching latest schema for subject 'debezium.tpcch.supplier-value' from registry: subject not found
[2020-10-14T04:04:41Z WARN  peeker] error trying to create source nation: db error: ERROR: fetching latest schema for subject 'debezium.tpcch.nation-value' from registry: subject not found    
[2020-10-14T04:04:41Z WARN  peeker] error trying to create source region: db error: ERROR: fetching latest schema for subject 'debezium.tpcch.region-value' from registry: subject not found                                                   
[2020-10-14T04:04:41Z WARN  peeker] error trying to create source supplier: db error: ERROR: fetching latest schema for subject 'debezium.tpcch.supplier-value' from registry: subject not found                                               
[2020-10-14T04:04:44Z WARN  peeker] Some sources were not successfully created! created=["customer", "neworder", "order", "orderline", "history", "district", "warehouse", "item"] failed=["nation", "stock", "supplier", "nation", "region", "supplier"]
[2020-10-14T04:04:44Z INFO  peeker] started 1 peek threads for 10 queries                                                                                                                                                                      
thread '<unnamed>' panicked at 'should be able to prepare a query: Error { kind: Db, cause: Some(DbError { severity: "ERROR", parsed_severity: None, code: SqlState("XX000"), message: "unknown catalog item \'q01\'", detail: None, hint: None, position: None, where_: None, schema: None, table: None, column: None, datatype: None, constraint: None, file: None, line: None, routine: None }) }', src/peeker/src/main.rs:119:26
```

And here are the logs after I ran `docker restart <container>`:

```
[2020-10-14T04:29:53Z INFO  peeker::args] substituting config var ENV_VAR w/ default val default
[2020-10-14T04:29:53Z INFO  peeker::args] substituting config var KAFKA_HOST w/ env var kafka
[2020-10-14T04:29:53Z INFO  peeker::args] substituting config var KAFKA_EXTERNAL_PORT w/ env var 9092
[2020-10-14T04:29:53Z INFO  peeker::args] substituting config var SR_HOST w/ env var schema-registry
[2020-10-14T04:29:53Z INFO  peeker::args] substituting config var OLAP_THREADS w/ env var 1
[2020-10-14T04:29:53Z INFO  peeker::args] substituting config var OLAP_THREADS w/ env var 1
[2020-10-14T04:29:53Z INFO  peeker] startup 2020-10-14 04:29:53.898388310 UTC
[2020-10-14T04:29:53Z INFO  peeker] serving prometheus metrics on port 16875
[2020-10-14T04:29:53Z INFO  peeker] Allowing chbench to warm up for 0 seconds at 2020-10-14 04:29:53.902997288 UTC
[2020-10-14T04:29:53Z INFO  peeker] Done warming up at 2020-10-14 04:29:53.903046133 UTC
[2020-10-14T04:29:53Z INFO  peeker] Deleted source in preparation for creation customer
[2020-10-14T04:29:54Z INFO  peeker] installed source customer for topic debezium.tpcch.customer
[2020-10-14T04:29:54Z INFO  peeker] Deleted source in preparation for creation item
[2020-10-14T04:29:54Z INFO  peeker] installed source item for topic debezium.tpcch.item
[2020-10-14T04:29:54Z INFO  peeker] installed source nation for topic debezium.tpcch.nation
[2020-10-14T04:29:54Z INFO  peeker] Deleted source in preparation for creation neworder
[2020-10-14T04:29:54Z INFO  peeker] installed source neworder for topic debezium.tpcch.neworder
[2020-10-14T04:29:54Z INFO  peeker] Deleted source in preparation for creation order
[2020-10-14T04:29:54Z INFO  peeker] installed source order for topic debezium.tpcch.order
[2020-10-14T04:29:54Z INFO  peeker] Deleted source in preparation for creation orderline
[2020-10-14T04:29:54Z INFO  peeker] installed source orderline for topic debezium.tpcch.orderline
[2020-10-14T04:29:55Z INFO  peeker] installed source stock for topic debezium.tpcch.stock
[2020-10-14T04:29:55Z INFO  peeker] installed source supplier for topic debezium.tpcch.supplier
[2020-10-14T04:29:55Z INFO  peeker] Deleted source in preparation for creation history
[2020-10-14T04:29:55Z INFO  peeker] installed source history for topic debezium.tpcch.history
[2020-10-14T04:29:55Z INFO  peeker] Deleted source in preparation for creation district
[2020-10-14T04:29:55Z INFO  peeker] installed source district for topic debezium.tpcch.district
[2020-10-14T04:29:55Z INFO  peeker] Deleted source in preparation for creation nation
[2020-10-14T04:29:55Z INFO  peeker] installed source nation for topic debezium.tpcch.nation
[2020-10-14T04:29:55Z INFO  peeker] installed source region for topic debezium.tpcch.region
[2020-10-14T04:29:55Z INFO  peeker] Deleted source in preparation for creation supplier
[2020-10-14T04:29:56Z INFO  peeker] installed source supplier for topic debezium.tpcch.supplier
[2020-10-14T04:29:56Z INFO  peeker] Deleted source in preparation for creation warehouse
[2020-10-14T04:29:56Z INFO  peeker] installed source warehouse for topic debezium.tpcch.warehouse
[2020-10-14T04:29:56Z INFO  peeker] installed view q01
[2020-10-14T04:29:56Z INFO  peeker] installed view q02
[2020-10-14T04:29:56Z INFO  peeker] installed view q05
[2020-10-14T04:29:56Z INFO  peeker] installed view q06
[2020-10-14T04:29:57Z INFO  peeker] installed view q08
[2020-10-14T04:29:57Z INFO  peeker] installed view q09
[2020-10-14T04:29:57Z INFO  peeker] installed view q12
[2020-10-14T04:29:57Z INFO  peeker] installed view q14
[2020-10-14T04:29:57Z INFO  peeker] installed view q17
[2020-10-14T04:29:57Z INFO  peeker] installed view q19
[2020-10-14T04:29:57Z INFO  peeker] started 1 peek threads for 10 queries
[2020-10-14T04:29:57Z INFO  peeker] Will terminate thread 0.0 at 2156-11-20T10:58:12.914533437Z
[2020-10-14T04:29:58Z WARN  peeker] for q05: db error: ERROR: At least one input has no complete timestamps yet: [User(58)]. Sleeping for 250ms
[2020-10-14T04:30:58Z INFO  peeker] peeked loadtest 169 times with 1 errors
```

**Comment by cuongdo** [comment]
I've noticed various race conditions with peeker. That code may need a closer look.

**Comment by cirego** [comment]
This was fixed by #6421.
